litgpt/model.py

```diff
@@ -324,6 +330,15 @@ def forward(
         k = k.view(B, T, self.config.n_query_groups, self.config.head_size)  # (B, T, nh_k, hs)
         v = v.view(B, T, self.config.n_query_groups, self.config.head_size)  # (B, T, nh_v, hs)
 
+        if self.config.norm_qk:
+            q = q.reshape(B, T, -1)  # (B, T, nh_q * hs)
+            q = self.norm_q(q)
+            q = q.view(B, T, self.config.n_head, self.config.head_size)
+
+            k = k.reshape(B, T, -1)  # (B, T, nh_k * hs)
+            k = self.norm_k(k)
+            k = k.view(B, T, self.config.n_query_groups, self.config.head_size)
+
```

Comment:
Is there a reason why you do this normalization here and not right after the `qkv.split`?
If you move it there, you will not have to do .reshape and .view again.